### PR TITLE
CSG-818: Add Advanced Datatable Component to Comet Extras

### PR DESCRIPTION
**Pull request by jbouder**
Add a more flexible and functional react data table to Comet

## Description

- Added [TanStack Table](https://tanstack.com/table/v8) package to project
- Added custom component based on TanStack table to Comet Extras, providing flexible data rendering, sorting, and paging
- Added Stories for new Data Table to Extras section
- Added Unit tests for coverage

## Related Issue

N/A

## Motivation and Context

- To provide a more customized data table for use in projects

## How Has This Been Tested?

- Local Testing
- Unit Testing
- Verified no accessibility issues

## Screenshots (if appropriate):
Screenshot 2023-09-05 at 11 44 52 AM
Screenshot 2023-09-05 at 11 44 36 AM
Screenshot 2023-09-05 at 3 03 01 PM
Screenshot 2023-09-05 at 11 51 38 AM

### CODE REVIEW

**Review by bapplejax — APPROVED** [review]
Looks good!

File: packages/comet-extras/src/components/data-table/data-table.tsx [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+  return (
+    <>
+      <table id={id} className={classNames('data-table', { striped }, className)}>
+        <thead>
+          {table.getHeaderGroups().map((headerGroup) => (
+            <tr key={headerGroup.id}>
+              {headerGroup.headers.map((header) => (
+                <th key={header.id}>
+                  <div
+                    {...{
```
Comment (LuisdelaVega) [inline packages/comet-extras/src/components/data-table/data-table.tsx:104]: Interesting choice. You're creating an object just to destructure it. Why not just keep it simple?

```tsx
<div
  className={header.column.getCanSort() ? 'cursor-pointer select-none' : ''}
  onClick={header.column.getToggleSortingHandler()}
>
```

**Review by LuisdelaVega — CHANGES_REQUESTED** [review]
Solid work! There's one thing I think we should address. See comments.

File: packages/comet-extras/src/components/data-table/data-table.tsx (hunk 1 above)
Comment (jbouder) [reply packages/comet-extras/src/components/data-table/data-table.tsx]: Good catch. I believe this was left over from when this section had additional configurations. Removed

File: packages/comet-extras/src/components/data-table/data-table.tsx [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+        <div className="table-paging">
+          <button
+            id="table-paging-prev-btn"
+            className="table-paging-btn table-paging-prev"
+            onClick={() => table.previousPage()}
+            disabled={!table.getCanPreviousPage()}
+          >
+            {'<'}
+          </button>
+          {Array.from({ length: table.getPageCount() }).map((row, index) => {
```
[the remark this answers is not in the document]
Comment (jbouder) [reply packages/comet-extras/src/components/data-table/data-table.tsx]: Updated as requested

**Pull request merged**